### PR TITLE
Add missing attributes and tcp connection open / close metrics

### PR DESCRIPTION
**Pull request by bianpengyuan**
cc @douglas-reid @geeknoid

### CODE REVIEW

File: content/docs/reference/config/policy-and-telemetry/attribute-vocabulary/index.md [hunk 1]
```diff
@@ -81,6 +81,7 @@ deployments will have agents (Envoy or Mixer adapters) that produce these attrib
 | `context.time`          | timestamp | The timestamp of Mixer operation. | |
 | `context.reporter.kind` | string | Contextualizes the reported attribute set. Set to `inbound` for the server-side calls from sidecars and `outbound` for the client-side calls from sidecars and gateways | `inbound` |
 | `context.reporter.uid`  | string | Platform-specific identifier of the attribute reporter. | `kubernetes://my-svc-234443-5sffe.my-namespace` |
+| `context.proxy_error_code` | string | Additional details about the response or connection from envoy. See %RESPONSE_FLAGS% in [envoy access log](https://www.envoyproxy.io/docs/envoy/latest/configuration/access_log#configuration) for more detail | `UH` |
```
Comment (geeknoid) [inline content/docs/reference/config/policy-and-telemetry/attribute-vocabulary/index.md]: envoy -> Envoy

I'm not crazy about the fact our attributes are leaking Envoy-specific state. That stuff needs to be neutral relative to the specific proxy in use if possible.

File: content/docs/reference/config/policy-and-telemetry/metrics/index.md [hunk 2]
```diff
@@ -151,3 +157,11 @@ We will describe metrics first and then the labels for each metric.
     {{< text yaml >}}
     connection_security_policy: conditional((context.reporter.kind | "inbound") == "outbound", "unknown", conditional(connection.mtls | false, "mutual_tls", "none"))
     {{< /text >}}
+
+*   **Response Flags**: Additional details about the response or connection from envoy.
+    See %RESPONSE_FLAGS% in [envoy access log](https://www.envoyproxy.io/docs/envoy/latest/configuration/access_log#configuration)
```
Comment (geeknoid) [inline content/docs/reference/config/policy-and-telemetry/metrics/index.md]: envoy -> Envoy

Same comment about having close coupling with Envoy...

File: content/docs/reference/config/policy-and-telemetry/attribute-vocabulary/index.md (hunk 1 above)
Comment (bianpengyuan) [reply content/docs/reference/config/policy-and-telemetry/attribute-vocabulary/index.md]: Reworded to make it more neutral. Thanks!

File: content/docs/reference/config/policy-and-telemetry/metrics/index.md (hunk 2 above)
Comment (bianpengyuan) [reply content/docs/reference/config/policy-and-telemetry/metrics/index.md]: Done.

File: .spelling [hunk 3]
```diff
@@ -527,6 +527,7 @@ Yessenov
 yournamespace
 Zack
 Zipkin
+RESPONSE_FLAGS
```
Comment (geeknoid) [inline .spelling]: Instead of putting this in the dictionary, please surround uses of this string with backticks

Comment (bianpengyuan) [reply .spelling]: Done.

**Pull request merged**